doc/ref_model/chapters/chapter03.md

```diff
@@ -128,7 +129,20 @@ _**Example**: a virtual compute descriptor as defined in TOSCA Simple Profile fo
 | `subnet` | network address of the subnet |
 | `acceleration` | key/value pairs for selection of the appropriate acceleration technology |
 
-<p align="center"><b>Table 3-4:</b> Attributes of network resources.</p>
+<p align="center"><b>Table 3-4:</b> Attributes of network resources</p>
+
+<a name="3.2.5"></a>
+### 3.2.5 Availability Zone
+An Availability zone is a logical grouping of resources (e.g. compute, block storage, network).  These logical groups allow cloud operators to segment their physical resources pools based on factors such as location (rack, datacenter), or indirect failure domain dependencies like power sources.  Workloads can leverage availability zones to avoid sharing failure domains for a workload, and thus increase its fault-tolerance.
```

Comment:
"An Availability zone is a logical grouping of resources" can we change to ""An Availability zone is a logical pool of  physical resources"? 

Do we have to state "(compute, block storage, network)'? We should not want to give the impression that each Availability Zone (AZ) will have their own storage and network.

"These logical groups allow cloud operators to segment their physical resources pools based on factors such as location (rack, datacenter), or indirect failure domain dependencies like power sources." can we change to ""These logical pools segment their physical resources of a cloud based on factors chosen by the cloud operator. The cloud operator may create availability zones based on location (rack, datacenter), or indirect failure domain dependencies like power sources."

Reply:
@pgoyal01 I agree with points 1 and 3, but don't understand the middle one.  Why wouldn't an AZ have it's own storage or network?  Surely that's what we want, to reduce the failure domain when an AZ fails?

Reply:
@tomkivlin I should have been clearer - we can't require the "AZ" failure domain in case of networks and storage as operators may support high availability of these for various reasons.  For tenant workloads on computes, AZ is an appropriate failure domain mechanism.

Reply:
@pgoyal01 I know that the block store is already going to be replicated or have some sort of redundancy scheme, but there are still issues of failure domains in terms of storage control plane, drivers, etc that can happen. 
There are several kinds of workloads (e.g. Kafka, clustered databases, etc) that replicate data amongst multiple nodes in a cluster to create a higher level of availability.  Having compute availability zones helps in that the DB instances can be divided so that they do not share fault domains at the compute level, but that does not address the issue of if there are shared fault scenarios in the storage tier.  Storage availability zones is one way to identify pools of storage that should not fail simultaneously, thus presenting a useful capability for these replicated/clustered data services.

Reply:
@kedmison I know at the RM we need to abstract away from implementation details but here is the scenario with OpenStack. 
1. In OpenStack unlike some public clouds, AZs do not isolate subnets.
1. An AZ is a logical partition of either compute, block storage or networking services where each of the services implements AZs differently.
1. A compute node can be in multiple host aggregates but in only one compute AZ.
1. Block Storage (Cinder) AZs may be configured for different backends. Many compute AZs can map to a Cinder AZ.
1. Neutron AZs (relatively new) allows grouping of network nodes that run networking services such as DHCP, L3, and others.

Compute AZs are more commonly used and, typically, are the subject when discussing AZs in the OpenStack context; unless referring to Cinder or Neutron AZs.

The failure domain for a typical AZ (a host aggregate that is visible to the tenants) depends upon implementations: it can be one or compute nodes or a rack or ...

Reply:
@pgoyal01 thanks for the discussion.  I'll take some of your points, one-by-one.
 
> 2. An AZ is a logical partition of either compute, block storage or networking services where each of the services implements AZs differently.

Openstack breaks it down this way, yes.  But is that the most useful way to present it to the users?  Perhaps we should be using Openstack's constructs to create a new presentation instead of passing them through directly.    Similar to how a 'container' isn't a first-class thing in the linux kernel, but instead an abstract entity composed of several cgroup implementations applied in a co-ordinated fashion.  An AZ in the datacenter space should be made up of compute separation (nova AZ) + storage separation (cinder AZ) + network isolation (independent physical network infrastructure as far up the leaf/spine/core-router tree as is possible).

> 3. A compute node can be in multiple host aggregates but in only one compute AZ.

Agree; no problem here.

> 4. Block Storage (Cinder) AZs may be configured for different backends. Many compute AZs can map to a Cinder AZ.

Here I have an observation: With a single Cinder AZ, there is likely a single Cinder back-end.  That can fail (not just a single-disk failure or single-storage-node failure but a total failure of the storage front-end.  This can cause all workloads using that storage to fail.  A properly designed workload could survive that event, but only if it was aware of and could leverage separate Cinder back-ends through Cinder AZ separation.  I think this is a useful addition to the availability model of an NFVI to have independent storage backends wherever there are independent compute AZs.

> The failure domain for a typical AZ (a host aggregate that is visible to the tenants) depends upon implementations: it can be one or compute nodes or a rack or ...

Agreed.  And, I understand that different operators may choose to implement failure domain separation strategies (e.g. different rooms w/ different HVAC services, and/or different utility power feeds, etc etc.)   But we should seriously look at ensuring that an AZ at least covers the compute domain and the storage domain.

Reply:
@kedmison we need separate requirements for the compute and storage domains including what are the characteristics (requirements) for a "failure domain" and then have the operators review and approve these requirements.

BTW these requirements are likely to impact both capital and operational expense.

Reply:
> the bottom line here, we need to shed light on the AZ in RM with the above @pgoyal01 modified definition "with the operator implementation strategy

@asawwaf @pgoyal01 this is a great discussion, and is one where we are really getting to some of the fine points and hard balances at the VNF<->NFVI interface.

@pgoyal01 Can I propose an option here?  That instead of having a single count of AZ in Table 4.1 in https://github.com/cntt-n/CNTT/pull/1250/files#diff-ea572522f5dd14982c918049c9c7dfa1R80 we update this to be a count of compute AZ and then add a count of storage AZ?  
the intent with this is to offer the flexibility to, for larger data centers, provide the full range of availability-zone capabilities while enabling smaller topologies, heading out towards the edge, to offer a reduced set.  This would also let operators make their own decisions while still providing clear information up-front to VNF vendors about what capabilities are present and can be leveraged.

Reply:
@kedmison Yes separating the different AZ counts would help. I still have a mental block.  If the operator is free to define an AZ on a non-failure domain (e.g., rack -- no independent power) then "Workloads can leverage availability zones" for redundancy (and for clusters) but not "avoid sharing failure domains for a workload, and thus increase its fault-tolerance." Can the sentences be changed to reflect that.

Reply:
I've been trying to leave it as an operator/deployment decision as to which specific failure domains are separated for compute AZs and storage AZs, as it's really a function of the size of deployment.
For a single rack deployment, one compute AZ and one storage AZ may be all that is provided.
For a multi-rack deployment, the AZs may not share TOR switches, but or PDUs, but still share building circuits, utility power, HVAC, buildings etc.
For major datacenters, the AZs may extend to having separate utility power feeds, separate buildings, etc.  

Simply providing the AZs can help with redundancy (which helps with fault tolerance), but by careful partitioning of failure domains along AZ boundaries, the NFVI deployment can help increase fault tolerance even further.